src/components/shapes/draw.js

```diff
@@ -605,13 +607,30 @@ function drawLabel(gd, index, options, shapeGroup) {
     // Remove existing label
     shapeGroup.selectAll('.shape-label').remove();
 
-    // If no label, return
-    if(!options.label.text) return;
+    // If no label text or texttemplate, return
+    if(!(options.label.text || options.label.texttemplate)) return;
+
+    // Text template overrides text
+    var text = options.label.text;
+    if(options.label.texttemplate) {
+        var templateValues = {};
+        if(options.type !== 'path') {
+            var _xa = Axes.getFromId(gd, options.xref);
+            var _ya = Axes.getFromId(gd, options.yref);
+            Object.keys(shapeLabelTexttemplateVars).forEach(function(key) {
+                var val = shapeLabelTexttemplateVars[key](options, _xa, _ya);
+                if(val !== undefined) templateValues[key] = val;
+            });
```

Comment:
```suggestion
            for (var key in shapeLabelTexttemplateVars) {
                var val = shapeLabelTexttemplateVars[key](options, _xa, _ya);
                if(val !== undefined) templateValues[key] = val;
            }
```
A little simpler and faster 😎